### PR TITLE
Retrieve system code from runbook

### PR DESCRIPTION
**Pull request by fenglish**
## Why?
If a repo contains system a and system b, and a change event is sent for system a, should both runbooks be updated?
[The trello ticket](https://trello.com/c/qxpEVk2X/71-clarify-relationship-between-change-api-events-and-runbook-updates-when-multiple-runbooks-exist-in-1-repo)

## What has changed?

- Retrieve systemCode from runbook.md

  **`systemCode` validate priority**
   1. System code from runbook.md contents
**(👆 this part is added by this PR to deal multiple runbooks)**
   2. From .github/runbooks.yml config, if it exists (this is shared with github app)
   3. From file name system-code_runbook.md (convention)
   4. the one in the Change API message

  
- Add an api call to check whether the systemCode exists in BizOps before update

## Scope and particulars of this PR (Please tick all that apply)

-   [ ] Tech hygiene (dependency updating & other tech debt)
-   [ ] Bug fix
-   [x] Feature
-   [ ] Documentation / Runbook update
-   [ ] Breaking change
-   [ ] Minor change (e.g. fixing a typo, adding config)
-   [ ] Review not required

## Reviewer checklist

-   [ ] Check this box if you feel you fully understand this PR
-   [ ] Has the documentation been updated (if applicable)?

### CODE REVIEW

File: lambdas/ingester/src/commands/ingest.js [hunk 1]
```diff
@@ -26,13 +27,28 @@ const decorateError = props => {
...
+		let message;
+		let code;
+		if (e.status === 404) {
+			message =
+				'Parse & validation complete. Biz-Ops update skipped (systemCode not exist in BizOps).';
+			code = 'parse-ok-systemCode-not-exist';
+		} else {
+			message =
+				"Parse & validation complete. Biz-Ops update skipped (BizOps api error when check systemCode's existence).";
+			code = 'parse-ok-biz-ops-api-error';
```
Comment (doramatadora) [inline lambdas/ingester/src/commands/ingest.js]: What do you think about:
-  `Biz-Ops update skipped: system code not found.` & `parse-ok-system-code-not-found`
-  `Parse & validation complete. Biz-Ops update skipped. Error from Biz-Ops: {e.message}.` – expose the specifics of the error from Biz-Ops API

Comment (ysugimoto) [reply lambdas/ingester/src/commands/ingest.js]: I think exposing the error from Biz-Ops API would be better.

When we call the API with an invalid or not found system code, the Biz-Ops API responds with an error message like `System [system-code] does not exist` . it is easy to catch what system code we sent to Biz-Ops API.

File: lambdas/ingester/src/commands/ingest.js [hunk 2]
```diff
@@ -3,6 +3,7 @@ const runbookMd = require('../lib/parser');
 const { validate, updateBizOps } = require('../lib/external-apis');
 const { updateSystemRepository } = require('../lib/biz-ops-client');
 const { transformCodesIntoNestedData } = require('../lib/code-validation');
+const { readSystem } = require('../lib/biz-ops-client');
```
Comment (doramatadora) [inline lambdas/ingester/src/commands/ingest.js]: Oooh just noticed, can this be rolled into the import on line 4, so we don't require the file twice?

Comment (fenglish) [reply lambdas/ingester/src/commands/ingest.js]: Oh I see! Thank you! I will do that

**Pull request merged**